bindings/rascal/utils/cg_utils.py

```diff
@@ -0,0 +1,340 @@
+"""
+Utilities to manipulate CG coefficients, rotations, and equivariant construction in general.
+Requires sympy to compute relevant coefficients.
+"""
+
+import numpy as np
+from scipy.spatial.transform import Rotation
+from ..neighbourlist.structure_manager import is_ase_Atoms
+
+
+# Just a few wrappers for sympy/scipy utility functions
+def _wigner_d(l, alpha, beta, gamma):
+    try:
+        from sympy.physics.wigner import wigner_d
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError(
+            "Calculation of Wigner D matrices requires a sympy installation"
+        )
+    return np.complex128(wigner_d(l, alpha, beta, gamma))
+
+
+def _rotation(alpha, beta, gamma):
+    return Rotation.from_euler("ZYZ", [alpha, beta, gamma]).as_matrix()
+
+
+def _cg(l1, l2, L):
+    try:
+        from sympy.physics.quantum.cg import CG
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError(
+            "Calculation of Clebsch-Gordan coefficients requires a sympy installation"
+        )
+
+    rcg = np.zeros((2 * l1 + 1, 2 * l2 + 1, 2 * L + 1), dtype=np.double)
+    if np.abs(l1 - l2) > L or np.abs(l1 + l2) < L:
+        return rcg
+    for m1 in range(-l1, l1 + 1):
+        for m2 in range(-l2, l2 + 1):
+            if np.abs(m1 + m2) > L:
+                continue
+            rcg[l1 + m1, l2 + m2, L + (m1 + m2)] += np.double(
+                CG(l1, m1, l2, m2, L, m1 + m2).doit()
+            )
+    return rcg
+
+
+# Convert from real to complex spherical harmonics (or corresponding
+# coefficients) and back. Uses the convention from Wikipedia
+isqrt2 = 1.0 / np.sqrt(2)
+sqrt2 = np.sqrt(2)
+
+
+def _r2c(sp):
+    l = (len(sp) - 1) // 2
+    rc = np.zeros(len(sp), dtype=np.complex128)
+    rc[l] = sp[l]
+    for m in range(1, l + 1):
+        rc[l + m] = (sp[l + m] + 1j * sp[l - m]) * isqrt2 * (-1) ** m
+        rc[l - m] = (sp[l + m] - 1j * sp[l - m]) * isqrt2
+    return rc
+
+
+def _c2r(cp):
+    l = (len(cp) - 1) // 2
+    rs = np.zeros(len(cp), dtype=np.float64)
+    rs[l] = np.real(cp[l])
+    for m in range(1, l + 1):
+        rs[l - m] = (-1) ** m * sqrt2 * np.imag(cp[l + m])
+        rs[l + m] = (-1) ** m * sqrt2 * np.real(cp[l + m])
+
+    return rs
+
+
+def spx_roll(spx, hypers):
```

Comment:
This name is not very clear, and it is a public function. How about `reshape_spherical_expansion`?

Also, I would take `n_max` & `l_max` explicitly as parameters instead of relying on hypers.

Reply:
as a compromise, it takes `max_radial` and `max_angular`, so you can also call it with **hypers

Reply:
👍